controllers/profiles/common/object_creators.go

```diff
@@ -218,10 +226,21 @@ func WorkflowPropsConfigMapCreator(workflow *operatorapi.SonataFlow) (client.Obj
 	return workflowproj.CreateNewAppPropsConfigMap(workflow, props), nil
 }
 
-func ConfigurePersistence(serviceContainer *corev1.Container, options *operatorapi.PersistenceOptions, defaultSchema, namespace string) *corev1.Container {
+func ConfigurePersistence(serviceContainer *corev1.Container, config *operatorapi.PersistenceOptions, defaultSchema, namespace string) (*corev1.Container, error) {
+	if config == nil {
+		return serviceContainer, nil
+	}
 	c := serviceContainer.DeepCopy()
-	if options != nil && options.PostgreSql != nil {
-		c.Env = append(c.Env, persistence.ConfigurePostgreSqlEnv(options.PostgreSql, defaultSchema, namespace)...)
+
+	var p *operatorapi.PersistencePostgreSQL
+	if config.PostgreSQL == nil && persistence.WorkflowConfig.GetPostgreSQLConfiguration() == nil {
+		return nil, fmt.Errorf("no persistence specification available in the workflow or in the platform")
+	}
+	if config.PostgreSQL != nil {
+		p = config.PostgreSQL
+	} else if persistence.WorkflowConfig.GetPostgreSQLConfiguration() != nil {
+		p = persistence.WorkflowConfig.GetPostgreSQLConfiguration()
```

Comment:
if we pick the platform configuration, yes, we must use the defaultSchema as it comes, basically the workflow name.